### PR TITLE
add autocomplete

### PR DESCRIPTION
**Pull request by DidenkoDenis306**
[DEMO LINK](https://DidenkoDenis306.github.io/react_autocomplete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,76 @@
...
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  useEffect(() => {
+    if (!query.trim()) {
+      setShowSuggestions(false);
+    }
+  }, [query]);
```
Comment (dffUqp) [inline src/App.tsx:34]: can we replace this useEffect with a variable like?🤔

```jsx
const isSuggestionsVisible = !query.trim() && showSuggestions;
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,76 @@
...
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const handleQueryOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (dffUqp) [inline src/App.tsx:41]:
```suggestion
  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,76 @@
...
+    setShowSuggestions(true);
+
+    const normalizePersonName = (human: Person) => {
+      return human.name.toLowerCase().trim();
+    };
+
+    return peopleFromServer.filter(
+      person => normalizePersonName(person)
+        .includes(appliedQuery.toLowerCase()),
+    );
```
Comment (dffUqp) [inline src/App.tsx:65]: let's remove normalizePersonName function.

```suggestion
    return peopleFromServer.filter(
      person => {
        const normalizedPersonName = person.name.toLowerCase().trim()
      
        return normalizedPersonName.includes(appliedQuery.toLowerCase())
     });
```

**Review by dffUqp — CHANGES_REQUESTED** [review]
LGTM:fire: but needs a few small fixes:upside_down_face:

File: src/App.tsx (hunk 1 above)
Comment (DidenkoDenis306) [reply src/App.tsx:34]: Unfortunately, it does not work

File: src/DropdownMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export const DropdownMenu: React.FC<Props> = ({
+  people,
+  onSelected,
+}) => {
+  return (
+    <div className="dropdown-menu" role="menu">
+      <div className="dropdown-content">
+        {people.length > 0
```
Comment (dffUqp) [inline src/DropdownMenu.tsx:16]:
```suggestion
        {!people.length
```

File: src/DropdownMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          ? (
+            people.map((person) => (
+              <div className="dropdown-item" key={person.slug}>
+                {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */}
+                <p
+                  className={person.sex === 'm'
+                    ? 'has-text-link'
+                    : 'has-text-danger'}
```
Comment (dffUqp) [inline src/DropdownMenu.tsx:24]: [classnames](https://www.npmjs.com/package/classnames  )

File: src/DropdownMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  onSelected,
+}) => {
+  return (
+    <div className="dropdown-menu" role="menu">
+      <div className="dropdown-content">
+        {people.length > 0
+          ? (
+            people.map((person) => (
+              <div className="dropdown-item" key={person.slug}>
+                {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */}
```
Comment (dffUqp) [inline src/DropdownMenu.tsx:20]: let's try to fix this eslint issue. you give `onClick`  prop to paragraph, I think its not a good idea.

File: src/App.tsx [hunk 7]
```diff
@@ -17,40 +77,17 @@ export const App: React.FC = () => {
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQueryOnChange}
```
Comment (dffUqp) [inline src/App.tsx:81]:
```suggestion
            onChange={handleQueryChange}
```

File: src/App.tsx (hunk 1 above)
Comment (dffUqp) [reply src/App.tsx:34]: try this
`const isSuggestionsVisible = !!query.trim() && showSuggestions;`

File: src/DropdownMenu.tsx (hunk 4 above)
Comment (DidenkoDenis306) [reply src/DropdownMenu.tsx:16]: Probably !!people.length? But elsint  says that's not the way to do it. So I think it's better to keep it that way.

**Pull request closed without merging**